src/deprecations.jl

```diff
@@ -0,0 +1,10 @@
+
+### v0.8 deprecations
+
+@deprecate ∇softmax(Δ, x; dims=1)   ∇softmax(Δ, x, softmax(x, dims=dims); dims=dims)
+@deprecate ∇softmax!(Δ, x; dims=1)   ∇softmax!(Δ, Δ, x, softmax(x, dims=dims); dims=dims)
+@deprecate ∇softmax!(out, Δ, x; dims=1)   ∇softmax!(out, Δ, x, softmax(x, dims=dims); dims=dims)
+
+@deprecate ∇logsoftmax(Δ, x; dims=1)   ∇logsoftmax(Δ, x, logsoftmax(x, dims=dims); dims=dims)
+@deprecate ∇logsoftmax!(Δ, x; dims=1)   ∇logsoftmax!(Δ, Δ, x, logsoftmax(x, dims=dims); dims=dims)
+@deprecate ∇logsoftmax!(out, Δ, x; dims=1)   ∇logsoftmax!(out, Δ, x, logsoftmax(x, dims=dims); dims=dims)
```

Comment:
I wonder whether these should really be deprecated. I could still see situations, where it's useful to just get the gradient without the forwards pass. My first thought was to maybe make `y` an optional keyword argument instead, but don't have a very strong opinion on this.

Reply:
Since `y` is the last argument, can't it simply have a default value? I mean permanently, just leave `∇softmax` having a 2-arg and 3-arg method.

Reply:
also @DhairyaLGandhi is in favor of the optional `y`, will change

Reply:
Yeah, sure, that's possible as well. I feel like a keyword argument might probably leave less ambiguity as to which argument specifies what, but would be fine with an optional positional argument as well.

Reply:
still, I'm not totally convinced: in many situations people could be computing the forward twice without noticing. I think that the benefit of a deprecation warn in these situations outweighs the slight annoyance of having to explicitly compute and pass the forward when needing only the gradient

Reply:
any further comment here? If someone has a strong opinion into keeping the 2-arg method I can reinstate it, otherwise I'd leave the deprecation

Reply:
I added back the 2-arg version